[CI] MixedClusterClientYamlTestSuiteIT test {p0=nodes.stats/11_indices_metrics/indices mappings exact count test for indices level} failing

**Build Scans:**
- [elasticsearch-intake #16455 / 8.18.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/pal4psfe6b7ju)
- [elasticsearch-pull-request #52920 / 8.18.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/5cw3mbnga4xzc)
- [elasticsearch-pull-request #52701 / 8.18.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/yfeyj3cu4nihq)

**Reproduction Line:**
```
./gradlew ":qa:mixed-cluster:v8.18.0#mixedClusterTest" -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=nodes.stats/11_indices_metrics/indices mappings exact count test for indices level}" -Dtests.seed=A87B0532A9233474 -Dtests.bwc=true -Dtests.locale=kea-CV -Dtests.timezone=Africa/Maseru -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bp0%3Dnodes.stats%2F11_indices_metrics%2Findices%20mappings%20exact%20count%20test%20for%20indices%20level%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [nodes.stats/11_indices_metrics:506]: 
Expected: <28>
     but: was <29>
```

**Issue Reasons:**
- [main] 3 failures in test test {p0=nodes.stats/11_indices_metrics/indices mappings exact count test for indices level} (1.4% fail rate in 211 executions)
- [main] 3 failures in step 8.18.0_bwc-snapshots (1.8% fail rate in 171 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (1.5% fail rate in 130 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -25,9 +25,6 @@ tests:
 - class: org.elasticsearch.xpack.ccr.FollowIndexSecurityIT
   method: testCleanShardFollowTaskAfterDeleteFollower
   issue: https://github.com/elastic/elasticsearch/issues/120339
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=nodes.stats/11_indices_metrics/indices mappings exact count test for indices level}
-  issue: https://github.com/elastic/elasticsearch/issues/120950
 - class: org.elasticsearch.action.admin.cluster.node.tasks.CancellableTasksIT
   method: testChildrenTasksCancelledOnTimeout
   issue: https://github.com/elastic/elasticsearch/issues/123568

